File: scala-play-server/app/api/RealTimeEventsApiImpl.scala
package api

import model.BadEventQueueIdError
import model.JsObject
import model.JsonSuccess
import model.JsonSuccessBase
import model.OneOfInvalidApiKeyErrorMissingArgumentErrorUserNotAuthorizedError
import model.OneOfRateLimitedError
import model.OneOfUserDeactivatedErrorRealmDeactivatedError

/**
  * Provides a default implementation for [[RealTimeEventsApi]].
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2021-08-08T20:45:49.166589Z[Etc/UTC]")
class RealTimeEventsApiImpl extends RealTimeEventsApi {
  /**
    * @inheritdoc
    */
  override def deleteQueue(queueId: String): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def getEvents(queueId: String, lastEventId: Option[Int], dontBlock: Option[Boolean]): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def realTimePost(eventTypes: Option[List[String]], narrow: Option[List[List[String]]], allPublicStreams: Option[Boolean]): Unit = {
    // TODO: Implement better logic

    
  }

  /**
    * @inheritdoc
    */
  override def registerQueue(applyMarkdown: Option[Boolean], clientGravatar: Option[Boolean], slimPresence: Option[Boolean], eventTypes: Option[List[String]], allPublicStreams: Option[Boolean], includeSubscribers: Option[Boolean], clientCapabilities: Option[JsObject], fetchEventTypes: Option[List[String]], narrow: Option[List[List[String]]]): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def restErrorHandling(): Unit = {
    // TODO: Implement better logic

    
  }
}

File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/api/RealTimeEventsApiImpl.java
package org.openapitools.vertxweb.server.api;

import org.openapitools.vertxweb.server.model.BadEventQueueIdError;
import org.openapitools.vertxweb.server.model.JsonSuccess;
import org.openapitools.vertxweb.server.model.JsonSuccessBase;
import org.openapitools.vertxweb.server.model.OneOfInvalidApiKeyErrorMissingArgumentErrorUserNotAuthorizedError;
import org.openapitools.vertxweb.server.model.OneOfRateLimitedError;
import org.openapitools.vertxweb.server.model.OneOfUserDeactivatedErrorRealmDeactivatedError;

import org.openapitools.vertxweb.server.ApiResponse;

import io.vertx.core.Future;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.handler.impl.HttpStatusException;

import java.util.List;
import java.util.Map;

// Implement this class

public class RealTimeEventsApiImpl implements RealTimeEventsApi {
    public Future<ApiResponse<JsonSuccess>> deleteQueue(String queueId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> getEvents(String queueId, Integer lastEventId, Boolean dontBlock) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<Void>> realTimePost(List<String> eventTypes, List<List<String>> narrow, Boolean allPublicStreams) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> registerQueue(Boolean applyMarkdown, Boolean clientGravatar, Boolean slimPresence, List<String> eventTypes, Boolean allPublicStreams, Boolean includeSubscribers, Object clientCapabilities, List<String> fetchEventTypes, List<List<String>> narrow) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<Void>> restErrorHandling() {
        return Future.failedFuture(new HttpStatusException(501));
    }

}
